e2e/e2etests/test_opcodes.go

```diff
@@ -0,0 +1,30 @@
+package e2etests
+
+import (
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/node/e2e/contracts/opcode"
+	"github.com/zeta-chain/node/e2e/runner"
+)
+
+func TestOpcodes(r *runner.E2ERunner, _ []string) {
+	// deploy the opcode contract and run function using opcode
+	addr, tx, opcodeCaller, err := opcode.DeployOpcode(r.ZEVMAuth, r.ZEVMClient)
+	require.NoError(r, err)
+	r.WaitForTxReceiptOnZEVM(tx)
+	r.Logger.Print("Deployed Opcode contract at %s", addr.Hex())
+
+	// check push 0
+	r.Logger.Print("Testing PUSH0 opcode...")
+	tx, err = opcodeCaller.TestPUSH0(r.ZEVMAuth)
+	require.NoError(r, err)
+	r.WaitForTxReceiptOnZEVM(tx)
+	r.Logger.Print("PUSH0 opcode verified")
```

Comment:
NIT : Do we need to print these ? or is info okay

Reply:
How `PUSH0` verification works? It seems that the bytecode used by this E2E test was generated by a recent compiler that supports `PUSH0`, and then it deploys the contract to `ZEVM` and calls `TestPUSH0`, this test is expected to pass if ZEVM is no older than Shanghai upgrade, otherwise it reverts?

Reply:
Yes it would revert if not supported

Reply:
It's a test to run manually so you want these type of logs